src/lib/components/Modal.svelte

```diff
@@ -1,31 +1,83 @@
 <script lang="ts">
-	import { fade } from 'svelte/transition';
+	import { modal } from '$lib/stores/modal';
+	import Icon from '$lib/components/Icon.svelte';
+
+	let element: HTMLDialogElement;
+
+	$: if (element) {
+		if ($modal) {
+			element.showModal();
+		} else {
+			element.close();
+		}
+	}
+
+	const closeOnBackdrop = (event: MouseEvent | KeyboardEvent) => {
+		if (event instanceof KeyboardEvent) return;
```

Comment:
Why do you listen for `KeyboardEvent` if you don't ever close the modal with the keyboard ?

Reply:
Because the listener is on the `dialog` html element so I need an on:keydown listener for accessibility purpose, but `clientY` & `clientX` doesn't exist on KeyboardEvent